### PR TITLE
Tag on-demand image by specified commit

### PR DESCRIPTION
**Pull request by AetherUnbound**
## Description

This is an issue that was identified while testing #644.

My understanding of what the various fields and available objects in a GitHub action was incorrect. What I thought was the _branch to tag and publish_ was actually the _branch the workflow is run from_. The side-effect of this was that the action would _always_ build `main`'s most recent CI/CD run rather than what we actually wanted to run.

This PR modifies the on-demand Docker image publishing so that it explicitly pulls the image built from a specific commit, and pushes said tagged image under the same commit.

I've also updated the documentation to include these changes.

## Testing Instructions

Try to use the workflow from this branch to build and tag this branch!

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: api/docs/guides/publish.md [hunk 1]
```diff
@@ -38,8 +41,11 @@ _Note: [GitHub's documentation provides screenshots for the various steps](https
...
-5. For the `image` parameter, specify either `api` or `ingestion_server`.
-6. Click the green "Run workflow" button.
+   1. (Optional) change the branch _that the workflow is running from_ (this does not change anything about the target commit)
+   2. For the `image` parameter, specify either `api` or `ingestion_server`.
+   3. For the `commit` parameter, specify the commit that the built Docker image artifact should be pulled from.
+5. Click the green "Run workflow" button.
 
 The newest run should appear in the Action list, and its status can be tracked from there.
+
+![Example Screenshot](/_static/publish_action_example.png)
```
Comment (stacimc) [inline api/docs/guides/publish.md:51]: Nice! Can we remove the [`TODO`](https://github.com/WordPress/openverse-api/blob/main/api/docs/guides/publish.md?plain=1#L36) to add screenshots or do you think there are more that would be helpful?

**Review by stacimc — APPROVED** [review]
Looks great and worked well in our manual tests in staging!

Screen Shot 2022-04-13 at 5 02 23 PM

File: api/docs/guides/publish.md (hunk 1 above)
Comment (AetherUnbound) [reply api/docs/guides/publish.md:51]: D'oh, we can! 😄

File: api/docs/guides/publish.md [hunk 2]
```diff
@@ -27,19 +27,23 @@ The workflow file for this action can be found here: [`push_docker_image.yml`](h
 
 This action **will only publish & tag a service's image with the commit SHA** (i.e. it will not tag with `latest` or the git reference name).
 This is useful for cases where one or more Docker images need to be tested on a staging environment before being merged and cut with a release.
-Note that this action can only run on branches/commits/tags that were made within the last 90 days [due to GitHub's default artifact retention policy](https://docs.github.com/en/organizations/managing-organization-settings/configuring-the-retention-period-for-github-actions-artifacts-and-logs-in-your-organization).
+This action also **requires that the CI/CD workflow has previously run on the specified commit**.
+This typically requires a PR in order to initiate.
+
+Note that this action can only run on commits that were made within the last 90 days [due to GitHub's default artifact retention policy](https://docs.github.com/en/organizations/managing-organization-settings/configuring-the-retention-period-for-github-actions-artifacts-and-logs-in-your-organization).
```
Comment (sarayourfriend) [inline api/docs/guides/publish.md:33]: Just a clarification here.

```suggestion
Note that this action can only run on commits that had successful CI/CD runs in the last 90 days [due to GitHub's default artifact retention policy](https://docs.github.com/en/organizations/managing-organization-settings/configuring-the-retention-period-for-github-actions-artifacts-and-logs-in-your-organization).
```

**Review by sarayourfriend — APPROVED** [review]
LGTM! Just one clarification bit for the docs I didn't catch before.

**Review by obulat — APPROVED** [review]

**Pull request merged**